src/main/scala/com.snowplowanalytics.iglu/ctl/IgluctlConfig.scala

```diff
@@ -21,18 +24,18 @@ case class IgluctlConfig(description: Option[String],
 object IgluctlConfig {
   /** Trait common to S3cp and Push commands */
   sealed trait IgluctlAction {
-    def process: Result
+    def process(httpClient: Client[IO]): Result
   }
 
   object IgluctlAction {
     case class S3Cp(command: Command.StaticS3Cp) extends IgluctlAction {
-      def process: Result =
+      def process(httpClient: Client[IO]): Result =
```

Comment:
It doesn't look like `process` does anything with `httpClient`. Should `httpClient` in the trait's `process` definition be optional with default `None`? I feel like it will clarify things, but the trade off is checking if there's a client in `Push.process`.